code/modules/projectiles/guns/specialist.dm

```diff
@@ -548,6 +548,20 @@ Note that this means that snipers will have a slowdown of 3, due to the scope
 	placed_overlay_iconstate = "pepper"
 
 //-------------------------------------------------------
+
+/obj/item/weapon/gun/rifle/pepperball/pepperball_mini	//TODO: firemode currently has to be changed before attaching the gun
+	name = "mini pepperball gun"						//TODO: gun can currently still be fired unattached
```

Comment:
GUN_ATTACHMENT_FIRE_ONLY under flags_gun_features to make sure it only fires when attached.